src/cargo/sources/path.rs

```diff
@@ -246,6 +260,52 @@ impl<'gctx> PathSource<'gctx> {
         Ok(None)
     }
 
+    /// Returns [`Some(gix::Repository)`](gix::Repository) if there is a sibling `Cargo.toml` and `.git`
```

Comment:
It's indeed copied, but it could be clearer as to what it really does. I have adjusted it (in the `gitoxide` version only for now) and is says:

```Rust
    /// Returns [`Some(gix::Repository)`](gix::Repository) if the discovered repository
    /// (searched upwards from `root`) contains a tracked `<root>/Cargo.toml`.
    /// Otherwise, the caller should fall back on full file list.
```